targets/csharp/source/PlayFabSDK/source/Qos/PlayFabQosApi.cs

```diff
@@ -0,0 +1,163 @@
+﻿namespace PlayFab.QoS
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
```

Comment:
I strongly discourage people from using Linq in code that the customer will use.
It invokes the garbage collector VERY heavily.
I know it makes "pretty code" and will often produce a result in fewer cpu cycles, but the cost is a huge GC hit later, resulting in an overall lower performance.
In my experience, ANY lazy by-hand-implementation can outperform Linq when you include the GC cost.